zcash_client_sqlite/src/wallet.rs

```diff
@@ -1478,9 +1478,9 @@ pub(crate) fn put_tx_meta<N>(
     )?;
 
     let tx_params = named_params![
-        ":txid": &tx.txid.as_ref()[..],
+        ":txid": &txid.as_ref()[..],
         ":block": u32::from(height),
```

Comment:
A test is failing because this insert fails due to no corresponding block existing in the `blocks` table. This presents a much larger problem, because we don't have any of the rest of the data needed to fill in the `blocks` table at the time we insert UTXOs (because none of it was necessary for spending UTXOs).

In a pairing, @nuttycom @daira and I decided that the current approach of this PR won't be workable for the mobile SDK 2.0.0 release, and instead we should alter the `v_transactions` view to make a bunch of its columns optional / best-effort (which is fine if the downstream users of it are only using it to display transaction history, and are not making any assumptions otherwise about the data).